watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Activities/MainActivity.java

```diff
@@ -148,6 +158,69 @@ public void onResume() {
         updateToolbarElevation();
     }
 
+    /**
+     * Gets the current registration ID for application on GCM service.
+     * If result is empty, the app needs to register.
+     *
+     * @return registration ID, or empty string if there is no existing
+     *         registration ID.
+     */
+    private String getRegistrationId() {
+        if (mRegistrationId.equals("none")) {
+            Log.i(TAG, "Registration not found.");
+            return "";
+        }
+
+        // Check if app was updated; if so, it must clear the registration ID
+        // since the existing registration ID is not guaranteed to work with
+        // the new app version.
+        int currentVersion = Utility.getAppVersion(this);
+        if (mAppVersion != currentVersion) {
+            Log.i(TAG, "App version changed.");
+            mPreferences.edit().putInt("app_version", currentVersion).commit();
+            return "";
+        }
+        return mRegistrationId;
+    }
+
+    /**
+     * Registers the application with GCM servers asynchronously.
+     * <p>
+     * Stores the registration ID and app versionCode in the application's
+     * shared preferences.
+     */
+    private void registerInBackground() {
```

Comment:
but couldn't that also go in the LandingPage? You should have access to the same version information there.